src/iarc7_motion/iarc_tasks/track_roomba_task.py

```diff
@@ -0,0 +1,168 @@
+
+        if self._state == TrackObjectTaskState.track:
+            try:
+                roomba_transform = self._tf_buffer.lookup_transform(
+                                    'level_quad',
+                                    self._roomba_id,
+                                    rospy.Time.now(),
+                                    rospy.Duration(self._TRANSFORM_TIMEOUT))
+            except (tf2_ros.LookupException,
+                    tf2_ros.ConnectivityException,
+                    tf2_ros.ExtrapolationException) as ex:
+                rospy.logerr('ObjectTrackTask: Exception when looking up transform')
+                rospy.logerr(ex.message)
+                return (TaskAborted(msg='Exception when looking up transform during roomba track'),)
+
+            #Creat point centered at drone's center
+            stamped_point = PointStamped()
+            stamped_point.point.x = 0
+            stamped_point.point.y = 0
+            stamped_point.point.z = 0
+
+            #returns point distances of roomba to center point of level quad
+            self._roomba_point =tf2_geometry_msgs.do_transform_point(
+                                                stamped_point, roomba_transform)
+
+            if not self._check_max_roomba_range():
+                return (TaskAborted(msg='The provided roomba is not found or not within a meter of the quad'),)
+
+            if self._check_min_roomba_range():
+                return (TaskDone(),)
+
+            #p-controller
+            x_error = (self._roomba_point.point.x * self._k_x + 
+                        self._roomba_odometry.twist.twist.linear.x)
+            y_error = (self._roomba_point.point.y * self._k_y + 
+                        self._roomba_odometry.twist.twist.linear.y)
+            z_error = self._z_holder.get_height_hold_response()
+
+            #caps x and y velocities
+            if abs(x_error) > self._MAX_TRANSLATION_SPEED:
+                x_error = x_error/abs(x_error) * self._MAX_TRANSLATION_SPEED
+
+            if abs(y_error) > self._MAX_TRANSLATION_SPEED:
+                y_error = y_error/abs(y_error) * self._MAX_TRANSLATION_SPEED
+
+            velocity = TwistStamped()
+            velocity.header.frame_id = 'level_quad'
+            velocity.header.stamp = rospy.Time.now()
+            velocity.twist.linear.x = x_error
+            velocity.twist.linear.y = y_error
+            velocity.twist.linear.z = z_error
+            
+            return (TaskRunning(), VelocityCommand(velocity)) 
+
+    #checks to see if passed in roomba id is available and
+    #that the drone and roomba are both within a 1 meter square
+    def _check_max_roomba_range(self):
+        for odometry in self._roomba_array.data:
+            if odometry.child_frame_id == self._roomba_id:
+                self._roomba_odometry = odometry
+                self._roomba_found =  True 
+        return (abs(self._roomba_point.point.x) < 1.0 and 
```

Comment:
Can this be changed to be a radius instead of a square. Can each term be on it's own line for readability?